client/packages/coldchain/src/Sensor/Components/SensorEditModal.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { FC, useState } from 'react';
+import { SensorFragment, useSensor } from '../api';
+import { useTranslation } from '@common/intl';
+import { useDialog } from '@common/hooks';
+import { DialogButton, useConfirmationModal } from '@common/components';
+import { ObjUtils } from '@common/utils';
+import { Grid } from '@openmsupply-client/common/src';
+import { EditableSensorTab } from './EditableSensorTab';
+import { NonEditableSensorTab } from './NonEditableSensorTab';
+
+interface SensorEditModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  sensor: SensorFragment;
+}
+
+interface useDraftSensorControl {
+  draft: SensorFragment;
+  onUpdate: (patch: Partial<SensorFragment>) => void;
+  onSave: () => Promise<void>;
+  isLoading: boolean;
+}
+
+const useDraftSensor = (seed: SensorFragment): useDraftSensorControl => {
+  const [sensor, setSensor] = useState<SensorFragment>({ ...seed });
+  const { mutate, isLoading } = useSensor.document.update();
+  const onUpdate = (patch: Partial<SensorFragment>) => {
+    setSensor({ ...sensor, ...patch });
+  };
+
+  const onSave = async () => mutate(sensor);
+
+  return {
+    draft: sensor,
+    onUpdate,
+    onSave,
+    isLoading,
+  };
+};
+
+export const SensorEditModal: FC<SensorEditModalProps> = ({
+  sensor,
+  isOpen,
+  onClose,
+}) => {
+  const t = useTranslation('coldchain');
+  const { Modal } = useDialog({ isOpen, onClose });
+  const { draft, onSave, onUpdate } = useDraftSensor(sensor);
+  const getConfirmation = useConfirmationModal({
+    title: t('heading.are-you-sure'),
+    message: t('message.conform-sensor-update'),
+  });
```

Comment:
I still have the confirmation modal here (modal on modal),

The UX felt a little strange without it so have left it in. 
Would it be better to remove?

Reply:
Stock details also has modal on modal, so I guess it's okay since it already exists